src/core/lexer.l

```diff
@@ -262,6 +262,10 @@ use[ \t\r\n]*"<"		{ BEGIN(cond_use); LOCATION_COUNT_LINES(parserlloc, yytext); }
 
 {UNICODE}+              { parser_error_pos -= strlen(yytext); return TOK_ERROR; }
 
+0x{H}{1,8}              {
```

Comment:
Note: There's an ugly bug in OpenSCAD's lexer which allows variables to start with a number. Your patch might break code which relies on that.

Reply:
It breaks anything that uses variables that would be legal hex constants.  It does not break variables that don't match the pattern.  Whether or not disallowing `0x0f` as a variable is acceptable is probably the biggest technical question in this change.

#3345 mentions a couple of possible alternative styles:  a function that accepts a hex string and returns a number, or a Python-style syntax like `h'0f'`.

Over in #3345 you will see that apparently @nophead relies on the identifiers-that-start-with-digits behavior.

Reply:
Yes but I don't think any of them look like hex constants.

Reply:
Still... allowing 0w123 and 0y123 as identifiers, but not allowing 0x123, would be odd.  Except for the incompatibility, I'd be in favor of disallowing identifiers that start with digits.  Being able to use real part numbers as names is nice, but if the part number is 4632-342 you can't use it, so maybe the starts-with-digit-otherwise-alphanumeric case is narrow enough to disallow.

Maybe we should think about some kind of pragma mechanism to enable/disable language features.  Start out with `#pragma hex-constants` to enable the new feature, but eventually migrate to `#pragma leading-digit-identifiers` to enable the legacy behavior.

Reply:
Instead of a per-feature pragma, I'd say just deprecate and use a monotonically increasing version pragma for backwards compatibility instead. That way, the language always moves forward, and it's much easier on us to manage. ..but that's a completely separate discussion :)